lib/src/folding/parsers/fallback.dart

```diff
@@ -13,21 +14,33 @@ class FallbackFoldableBlockParser extends TextFoldableBlockParser {
   final List<String> singleLineCommentSequences;
   final List<String> importPrefixes;
 
+  /// [ ['/*', '*/'] , ...]
+  final List<Tuple2<String, String>>? multilineCommentSequences;
+
   /// The size of a rolling window to remember processed characters.
   final int _tailLength;
 
+  String? _startedMultilineCommentWith;
+
   /// If in a string literal the last char was a backslash.
   bool _wasBackslash = false;
 
-  bool _isInSingleQuoteLiteral = false;
   bool _isInDoubleQuoteLiteral = false;
-  bool _foundServiceSingleLineComment = false;
+  bool _isInMultilineComment = false;
+  bool _isInSingleQuoteLiteral = false;
   bool _isLineStart = true;
+  bool _isPossibleCommentSequence = false;
+  bool _foundServiceSingleLineComment = false;
+  bool _shouldEndMultilineComment = false;
 
   FallbackFoldableBlockParser({
     required this.singleLineCommentSequences,
     required this.importPrefixes,
-  }) : _tailLength = singleLineCommentSequences.map((s) => s.length).max;
+    this.multilineCommentSequences,
```

Comment:
```suggestion
    this.multilineCommentSequences = [],
```
Then make it non-nullable